### DISCUSSION TITLE
txt2vid: dynamically change "Sampling Steps" based on a preview video render.

### DISCUSSION THREAD

**Discussion by cstueckrath** [hlky/stable-diffusion-webui#1076] [Feature suggestions]
currently txt2vid has two relevant sliders named Sampling Steps and Inference Steps.
What Inference Steps does is comparable to Sampling Steps in txt2img: Quality and detail of the image
What Sampling Steps does is to control the "frame rate" of the video. If you set this to 64 and render a video and then set it to 128 and render from the same seed, the resulting video will have 1 frame between each frame of the first video.
If you double the rendered frames, the beginning and end of the videos will be the same (+/- a frame)

We can use this to skip through a video and see, what it will look like before letting it run over night with high quality settings.
But if we could change the settings while running, we could  even cherrypick the part of the video we want.

1) run txt2vid in "preview mode": low sampling steps (2, 4, 8?), low quality inference steps (like 20 or so?)
2) show video in some kind of timeline, like images side by side
3) mark beginning and end frame of the target video
4) run the target render
  a) calculate frames at very low sampling steps and inference steps (1) to reach beginning frame
  b) iterate through until beginning frame is reached
  c) change sampling steps and inference steps mid flight to user configured values (e.g. 128/64)
  d) render until end frame is reached
5) drop frames before beginning frame and render video

**Comment by andzejsp** [comment]
Can i ask layman's question? How can i get this feature in the current web ui? I dont see any txt2vid panels / sliders/ settings in the UI. Maybe there is another .cmd file i need to run to get to it? I know there is a repo out there that focuses on the video generation but its a separate repo from this and its a bit clunky and not many options to tweak in the UI

**Reply by tehgringe** [reply]
You need to perhaps switch to the dev branch of the code and launch the web-streamlit.cmd as that appears to have then launch an instance of the webui with txt2vid enabled. This is dev stage, experimental and will come with zero support. Things will not work. So not allowed to complain.

**Reply by cstueckrath** [reply]
and you should join the Discord if running from dev branch. Most error discussions go there.
This feature request was discussed there. I just copied it here so it won't get lost.

**Reply by andzejsp** [reply]
thanks for the replies, but you cant assume that everyone has a discord. its free, but it is also evil. Anyway, ill test it later on my oilrig.

Also a side note. Is the repo moving from gradio to streamlit? A quick look at streamlit made me think its more UI friendly. I mean more responsive, and also you get the preview while you generate, thats just amazing.

While gradio UI feels like you post a request, and pray to gods that you get reply.

**Reply by cstueckrath** [reply]
I don't assume and it's certainly not required.
